sharktank/sharktank/tools/eval_llm_vmfb.py

```diff
@@ -47,58 +53,70 @@ def export_ir(irpa):
     return ir, config
 
 
-def compile_ir(ir, iree_hal_target_device, iree_hip_target):
-
+def compile_ir(ir, iree_hal_target_devices: list[str], iree_hip_target: str):
     logging.log(
-        logging.INFO, f"Compiling VMFB on {iree_hal_target_device} - {iree_hip_target}"
+        logging.INFO, f"Compiling VMFB on {iree_hal_target_devices} - {iree_hip_target}"
     )
     extra_args = [
-        f"--iree-hal-target-device={iree_hal_target_device}",
-        f"--iree-hip-target={iree_hip_target}",
+        f"--iree-hal-target-device={device}" for device in iree_hal_target_devices
     ]
+    extra_args += [f"--iree-hip-target={iree_hip_target}"]
     vmfb = iree.compiler.compile_str(ir, extra_args=extra_args)
     return vmfb
 
 
-def get_instance(vmfb, config, irpa, iree_hal_target_device, iree_hip_target):
+def get_instance(
+    vmfb: str | None,
+    config: str | None,
+    irpa: str,
+    iree_hal_target_devices: list[str] | None,
+    iree_hip_target: str | None,
+    pipeline_parallelism_size: int,
+) -> LlmInstance:
     if vmfb is None:
-        if iree_hal_target_device is None:
+        if iree_hal_target_devices is None:
             raise ValueError("--iree-hal-target-device is required")
 
         if iree_hip_target is None:
             raise ValueError("--iree-hip-target is required")
 
         if config is not None:
             raise ValueError("Config found without corresponding vmfb")
-        ir, config = export_ir(irpa)
-        vmfb = compile_ir(ir, iree_hal_target_device, iree_hip_target)
+        ir, config = export_ir(irpa, pipeline_parallelism_size)
+        vmfb = compile_ir(ir, iree_hal_target_devices, iree_hip_target)
 
     if isinstance(config, str):
-        config = ServiceConfig.load(config)
+        config: ServiceConfig = ServiceConfig.load(config)
+    assert pipeline_parallelism_size == len(
```

Comment:
You should assume that if the config is built and passed through it returns a valid config.